### PR TITLE
add HIP to the CI generator parameter

### PR DESCRIPTION
**Pull request by SimeonEhrig**
The Hip jobs are the first kind of jobs, which are automatically generated by the job generator. Because we test all supported CMake and Boost versions too, the initial amount of generated jobs is much more, than the number of hand written HIP jobs, which we used before. Because of the properties of the n-wise combination generator, if we add more compilers to the generator (like nvcc, gcc, clang as C++ and CUDA compiler), each new compiler will create less additional new jobs until we arrive a stable number of maximum jobs (depends on the number of combination parameters).

### CODE REVIEW

File: script/job_generator/alpaka_globals.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""This module contains constants used for the alpaka job generation.
```
Comment (j-stephan) [inline script/job_generator/alpaka_globals.py]: Shouldn't we have a copyright header here?

File: script/job_generator/alpaka_globals.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# additional alpaka specific parameters
+BUILD_TYPE: str = "build_type"
+TEST_TYPE: str = "test_type"
+
+# possible values of BUILD_TYPE
+CMAKE_RELEASE: str = "Release"
+CMAKE_DEBUG: str = "Debug"
+BUILD_TYPES: List[str] = [CMAKE_RELEASE, CMAKE_DEBUG]
+
+# possible values of TEST_TYPE
```
Comment (j-stephan) [inline script/job_generator/alpaka_globals.py:24]: What about analysis?

**Review by j-stephan — COMMENTED** [review]
Nice! Just two questions:

File: script/job_generator/alpaka_globals.py (hunk 2 above)
Comment (SimeonEhrig) [reply script/job_generator/alpaka_globals.py:24]: I only transformed the HIP jobs from hand written to auto generated. There was no analysis job. I would add it, if we transform a job, which is doing analysis.

File: script/job_generator/alpaka_globals.py (hunk 1 above)
Comment (SimeonEhrig) [reply script/job_generator/alpaka_globals.py]: I can add it.

**Comment by SimeonEhrig** [comment]
I will handle this 3 TODO's in a extra PR:

https://github.com/alpaka-group/alpaka/blob/076eb05bc1f49d06ee079e1724345bdb214d82e2/script/job_generator/generate_job_yaml.py#L171-L179

https://github.com/alpaka-group/alpaka/blob/076eb05bc1f49d06ee079e1724345bdb214d82e2/script/job_generator/generate_job_yaml.py#L245-L246

https://github.com/alpaka-group/alpaka/blob/076eb05bc1f49d06ee079e1724345bdb214d82e2/script/job_generator/generate_job_yaml.py#L247-L248

**Review by j-stephan — APPROVED** [review]

**Pull request merged**